dbms/src/Functions/FunctionsStringReplace.h

```diff
@@ -122,10 +122,40 @@ class FunctionStringReplace : public IFunction
 
         ColumnWithTypeAndName & column_result = block.getByPosition(result);
 
+        bool src_const = column_src->isColumnConst();
         bool needle_const = column_needle->isColumnConst();
         bool replacement_const = column_replacement->isColumnConst();
 
-        if (needle_const && replacement_const)
+        if (src_const){
+            if (!needle_const && replacement_const)
+            {
+                executeImplConstReplacement(
+                    column_src,
+                    column_needle,
+                    column_replacement,
+                    pos,
+                    occ,
+                    match_type,
+                    column_result
+                );
+            }else if (!needle_const && !replacement_const)
+            {
+                executeImplConstFirstParaReplacement(
+                    column_src,
+                    column_needle,
+                    column_replacement,
+                    pos,
+                    occ,
+                    match_type,
+                    column_result
+                );
+            }else
+            {
+                throw Exception(
+                    "UnImplement function.",
+                    ErrorCodes::BAD_ARGUMENTS);
+            }
```

Comment:
Possibly no need for a lot of impls because we are fixing for an edge case that only discovered in tests. It is acceptable to be not performance optimal.

Reply:
In my test case, some sql like :
```sql
explain SELECT /*+ read_from_storage(tiflash[my_table]) */ REPLACE('Hello World', my_table.col_11, my_table.col_14)  FROM my_table;
```
will get same error.  I think it may need to process, so I add these function to deal with such problems. 😂